include/swift/SILOptimizer/Analysis/RegionAnalysis.h

```diff
@@ -28,6 +28,12 @@ class RegionAnalysisValueMap;
 
 namespace regionanalysisimpl {
 
+#ifndef NDEBUG
+/// Global bool set only when asserts are enabled to ease debugging by causing
+/// unknown pattern errors to cause an assert so we drop into the debugger.
+extern bool AbortOnUnknownPatternMatchError;
+#endif
+
```

Comment:
The code in `TransferNonSendable.cpp` that uses `AbortOnUnknownPatternMatchError` is not guarded by `NDEBUG`.

https://github.com/swiftlang/swift/blob/3ccbcfe8cd40ded2d261b4011e1cc2e195361389/lib/SILOptimizer/Mandatory/TransferNonSendable.cpp#L681-L684

https://github.com/swiftlang/swift/blob/3ccbcfe8cd40ded2d261b4011e1cc2e195361389/lib/SILOptimizer/Mandatory/TransferNonSendable.cpp#L1098-L1101

https://github.com/swiftlang/swift/blob/3ccbcfe8cd40ded2d261b4011e1cc2e195361389/lib/SILOptimizer/Mandatory/TransferNonSendable.cpp#L1161-L1164

https://github.com/swiftlang/swift/blob/3ccbcfe8cd40ded2d261b4011e1cc2e195361389/lib/SILOptimizer/Mandatory/TransferNonSendable.cpp#L1651-L1654

Reply:
@drodriguez Thanks for the report! I am fixing it here: https://github.com/swiftlang/swift/pull/74911